### PR TITLE
ONE: Added card "Furnace Punisher" and its abilities and effects

### PR DESCRIPTION
**Pull request by Merlingilb**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/f/FurnacePunisher.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import mage.players.Player;
+
+import java.util.UUID;
+
+public class FurnacePunisher extends CardImpl {
+
+    private static final FilterLandPermanent filter = new FilterLandPermanent();
+
+    static {
+        filter.add(SuperType.BASIC.getPredicate());
```
Comment (PurpleCrowbar) [inline Mage.Sets/src/mage/cards/f/FurnacePunisher.java:27]: This filter is unnecessary as it is out of the scope of the relevant effect (which has its own identical copy of this filter) and is not used within this `FurnacePunisher` class

File: Mage.Sets/src/mage/cards/f/FurnacePunisher.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        //Menace
+        this.addAbility(new MenaceAbility(false));
+
+        //At the beginning of each player’s upkeep, Furnace Punisher deals 2 damage to that player unless they control
+        //two or more basic lands.
+        this.addAbility(new BeginningOfUpkeepTriggeredAbility(new FurnacePunisherEffect(),
+                TargetController.EACH_PLAYER,
+                false
+        ));
```
Comment (PurpleCrowbar) [inline Mage.Sets/src/mage/cards/f/FurnacePunisher.java:45]: Please apply `.setTriggerPhrase("At the beginning of each player's upkeep, ")` to `BeginningOfUpkeepTriggeredAbility` for desired rule text

**Review by PurpleCrowbar — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/f/FurnacePunisher.java (hunk 1 above)
Comment (Merlingilb) [reply Mage.Sets/src/mage/cards/f/FurnacePunisher.java]: done

File: Mage.Sets/src/mage/cards/f/FurnacePunisher.java (hunk 2 above)
Comment (Merlingilb) [reply Mage.Sets/src/mage/cards/f/FurnacePunisher.java]: done

**Pull request merged**